apps/addons/management/commands/sign_addons.py

```diff
@@ -1,33 +1,50 @@
 # -*- coding: utf-8 -*-
-import sys
+from optparse import make_option
 
 from django.core.management.base import BaseCommand, CommandError
 
-from addons.models import Addon
+import amo
+from files.models import File
+from lib.crypto.packaged import sign_file, SigningError
 
 
 class Command(BaseCommand):
-    args = '<addon_id addon_id ...>'
-    help = 'Sign a list of addons'
+    args = '[<addon_id addon_id ...>]'
+    help = 'Sign a list of addons, or all addons if none specified.'
+    option_list = BaseCommand.option_list + (
+        make_option('--resign', action='store_true',
+                    help=('Sign all files, even already signed ones'),
+                    default=False),
+        make_option('--noinput', action='store_false', dest='interactive',
+                    default=True,
+                    help="Do NOT prompt the user for input of any kind."),
+    )
 
     def handle(self, *args, **options):
-        if len(args) == 0:
-            raise CommandError('Please provide at least one addon ID')
-
-        for addon_id in args:
-            if not addon_id.isdigit():
-                sys.stderr.write(
-                    'Warning: Addon ID should be an integer (%s).\n'
-                    % addon_id)
-                continue
+        to_sign = File.objects.filter(filename__endswith='.xpi')
+        if not options['resign']:
+            to_sign = to_sign.exclude(
+                signed_status__in=[amo.SIGNED_FULL, amo.SIGNED_PRELIM])
 
+        if len(args) == 0:
+            if options['interactive']:
+                confirm = raw_input("You didn't provide any addon id, do you "
+                                    "confirm you want to sign them all (y/n)?")
+                if confirm != 'y':
+                    raise CommandError('Aborting')
+        else:
+            addon_ids = [int(addon_id) for addon_id in args]
+            to_sign = to_sign.filter(version__addon_id__in=addon_ids)
+
+        num_files = to_sign.count()
+        if not num_files:
+            raise CommandError('No files to sign, maybe try with --resign?')
+
+        self.stdout.write('Starting the signing of %s files' % num_files)
+        for file_obj in to_sign:
```

Comment:
feels like we should be using celery in here, but since running this should be rare it should be fine.